stdpopsim/models.py

```diff
@@ -162,6 +185,48 @@ def num_populations(self):
     def num_sampling_populations(self):
         return sum(int(pop.allow_samples) for pop in self.populations)
 
+    # Make populations a property to restrict the order of populations
+    @property
```

Comment:
See point below --- I think we can check both of these things as a standard test case, rather than doing it at run time.